purchase_sale_stock_inter_company/models/stock_move_line.py

```diff
@@ -0,0 +1,95 @@
+from odoo import api, models
+
+
+class StockMoveLine(models.Model):
+    _inherit = "stock.move.line"
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        new_move_lines = super().create(vals_list)
+        for move_line in new_move_lines:
+            purchase_line_origin = (
+                move_line.move_id.sale_line_id.sudo().auto_purchase_line_id
+            )
+            if move_line.state == "done" and purchase_line_origin:
+                po_moves = purchase_line_origin.move_ids.filtered(
```

Comment:
Is it assured here that only one `stock.move` is selected?

Reply:
You're right, now the moves are iterated and filtered by product. Thanks for the observation!